### PR TITLE
Release 0.31.0

### PR DESCRIPTION
**Pull request by colleend**
# Release Checklist
- [x] Updated CHANGELOG with latest changes
- [x] Updated `__VERSION__` in `semgrep/semgrep/__init__.py`
- [x] Updated install_requires version in `setup.py` for pre-commit in the repo root
- [x] Updated latest release and release date in `CHANGELOG.md`
- [x] Tests/CI passes on this Pull Request
- [x] Build Release Candidate finished successfully
- [x] Update semgrep action
- [x] Publish Github Release Page
- [x] Sanity check Pypi Packages
- [ ] Sanity check Homebrew core release
- [x] Sanity check docker images (X.X.X and latest tag)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,18 +4,34 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
...
+- Fix usage of '...' in special builtin arguments for PHP (#1963).
+- Fix automatic semicolon insertion parse error in javascript (#1960).
+
+### Added
+- kotlin-tree-sitter integration into semgrep-core. Can now call
+  dump-tree-sitter-cst on kotlin files.
+- c++ tree-sitter integration into semgrep-core (#1952).
+- More documents for language porting.
+- Error handling in spacegrep to print stderr when CalledProcessError occurs.
+- Scaning of files with no extension in generic mode.
```
Comment (brendongo) [inline CHANGELOG.md]: This is already on line 14
```suggestion
```

Comment (brendongo) [reply CHANGELOG.md]: Or could also remove line 14?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,18 +4,34 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
...
 
 ## [0.30.0](https://github.com/returntocorp/semgrep/releases/tag/v0.30.0) - 2020-11-03
 
 ### Added
 
 - Better examples for the generic mode aka spacegrep (#1951).
 
+
```
Comment (brendongo) [inline CHANGELOG.md]: Stray line?

Comment (colleend) [reply CHANGELOG.md]: ah nice catch

**Comment by brendongo** [comment]
Homebrew PR needs to be updated https://github.com/Homebrew/homebrew-core/pull/64518 but since we are pushing a 0.31.1 release I'll just make a new PR

**Pull request merged**